atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingKeyValueServiceImpl.java

```diff
@@ -17,50 +17,40 @@
 package com.palantir.atlasdb.transaction.impl.expectations;
 
 import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Multimap;
 import com.google.common.util.concurrent.Futures;
 import com.google.common.util.concurrent.ListenableFuture;
 import com.google.common.util.concurrent.MoreExecutors;
+import com.palantir.atlasdb.cell.api.AutoDelegate_TransactionKeyValueService;
+import com.palantir.atlasdb.cell.api.TransactionKeyValueService;
 import com.palantir.atlasdb.keyvalue.api.BatchColumnRangeSelection;
-import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
-import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
 import com.palantir.atlasdb.keyvalue.api.Cell;
 import com.palantir.atlasdb.keyvalue.api.ColumnRangeSelection;
 import com.palantir.atlasdb.keyvalue.api.ColumnSelection;
-import com.palantir.atlasdb.keyvalue.api.InsufficientConsistencyException;
-import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.keyvalue.api.RangeRequest;
 import com.palantir.atlasdb.keyvalue.api.RowColumnRangeIterator;
 import com.palantir.atlasdb.keyvalue.api.RowResult;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
 import com.palantir.atlasdb.keyvalue.api.Value;
-import com.palantir.atlasdb.keyvalue.impl.ForwardingKeyValueService;
 import com.palantir.atlasdb.transaction.api.expectations.TransactionReadInfo;
 import com.palantir.atlasdb.util.MeasuringUtils;
 import com.palantir.common.base.ClosableIterator;
-import com.palantir.common.exception.AtlasDbDependencyException;
-import com.palantir.logsafe.logger.SafeLogger;
-import com.palantir.logsafe.logger.SafeLoggerFactory;
 import com.palantir.util.paging.TokenBackedBasicResultsPage;
 import java.lang.reflect.Array;
-import java.util.List;
 import java.util.Map;
-import java.util.Set;
 import java.util.function.ToLongFunction;
 import one.util.streamex.EntryStream;
 
-public final class TrackingKeyValueServiceImpl extends ForwardingKeyValueService implements TrackingKeyValueService {
-    private static final SafeLogger log = SafeLoggerFactory.get(TrackingKeyValueServiceImpl.class);
-
-    private final KeyValueService delegate;
+public final class TrackingKeyValueServiceImpl
+        implements AutoDelegate_TransactionKeyValueService, TrackingKeyValueService {
```

Comment:
I have not added #getDelegates to TransactionKeyValueService, hence no Forwarding* implementation. Also ForwardingKeyValueService feels like a broken implementation to me.

Reply:
Those just end up proxying it correctly and computing a union. Anyway, I think we're in sync.